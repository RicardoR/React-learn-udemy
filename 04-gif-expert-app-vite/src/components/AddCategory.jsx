import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState();

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    };

    const onFormSubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim().length <= 3) {
            return;
        }

        onNewCategory(inputValue.trim());
        setInputValue('');
    };

    return (
        <form onSubmit={onFormSubmit}>
            <input
                type="text"
                placeholder="Search gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    );
};

