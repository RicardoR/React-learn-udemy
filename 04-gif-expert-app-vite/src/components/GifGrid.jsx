import PropTypes from 'prop-types';
import { getGifs } from '../helpers/getGifs';


export const GifGrid = ({category}) => {

    getGifs(category);

    return (
        <>
            <h3>{category}</h3>
        </>
    );
};


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};
