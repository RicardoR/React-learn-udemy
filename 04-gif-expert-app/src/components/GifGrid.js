import React, { useEffect, useState } from 'react';
import { apiKey, giphySearchEndPoint } from '../secret/giphy-data';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [imageList, setImages] = useState([]);

    useEffect(() => {
      getGifs();
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    
    const getGifs = async () => {
      const url = `${giphySearchEndPoint}?api_key=${apiKey}&q=${category}&limit=10`;
      const response = await fetch(url);
      const { data } = await response.json();
      
      const gifs = data.map(({ id, title, images }) => {
        return {
          id: id,
          title: title,
          url: images?.downsized_medium.url,
          };
        });
        setImages(gifs);
    };


  return (
    <>
      <h3>{category}</h3>
        {
        imageList.map((img) =>
          <GifGridItem
            {...img}
            key={img.id}
          />
        )
        }
    </>
  );
}
