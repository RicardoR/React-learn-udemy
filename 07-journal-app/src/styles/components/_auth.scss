.auth__main {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: $primary;
    margin: 0px;
    height: 100vh;
    width: 100vw;
}

.auth__box-container {
    background-color: white;
    box-shadow: 0px 3px $dark-grey;
    border-radius: 2px;
    padding: 20px;
    width: 250px;
}

.auth__title {
    color: darken($primary, 20%);
}

.auth__input {
    color: $dark-grey;
    border: 0px;
    border-bottom: 1px solid $light-grey;
    font-size: 16px;
    margin-bottom: 10px;
    height: 20px;
    width: 100%;
    transition: border-bottom .3s ease;

    &:focus {
        outline: none;
        border-bottom: 1px solid $primary;
    }
}

