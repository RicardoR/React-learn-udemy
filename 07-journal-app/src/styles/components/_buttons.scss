.btn {
    background-color: transparent;
    border: none;
    color: white;
    cursor: pointer;
    font-size: 12px;
    padding: 7px 10px 7px 10px;
    transition: color 0.3s ease-in-out;
    &:focus {
        outline: none;
    }

    &:hover {
        color: darken($color: white, $amount: 20);
    }
}

.btn-primary {
    background-color: $primary;
    border-radius: 2px;
    box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.25);
    transition: box-shadow .3s ease;
    &:hover {
        box-shadow: 0 0 6px #4285f4;
        background-color: darken($color: $primary, $amount: 15);
    }
   
    &:active {
        background: #1669f2;
    }
    &:disabled {
        background-color: lighten($color: $primary, $amount: 15);
    }
}

.btn-block {
    display: block;
    width: 100%;
}