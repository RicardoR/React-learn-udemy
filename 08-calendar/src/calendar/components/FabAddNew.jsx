import { addHours } from 'date-fns';
import { useCalendarStore, useUiStore } from '../../store/hooks';

export const FabAddNew = () => {
  const { openDateModal } = useUiStore();
  const { setActiveEvent } = useCalendarStore();
  const handleOpen = () => {
    setActiveEvent({
      title: null,
      start: new Date(),
      end: addHours(new Date(), 2),
      notes: null,
      bgColor: '#fafafa',
      user: {
        _id: null,
        name: null,
      },
    });

    openDateModal();
  };

  return (
    <button onClick={handleOpen} className="btn btn-primary fab">
      <i className="fas fa-plus"></i>
    </button>
  );
};
