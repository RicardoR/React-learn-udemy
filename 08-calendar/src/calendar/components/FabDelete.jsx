import { useCalendarStore } from '../../store/hooks';

export const FabDelete = () => {
  const { startDeletingEvent, hasEventSelected } = useCalendarStore();
  const handleDelete = () => startDeletingEvent();

  return (
    <button
      style={{
        display: hasEventSelected ? '' : 'none',
      }}
      onClick={handleDelete}
      className="btn btn-danger fab-danger"
    >
      <i className="fas fa-trash-alt"></i>
    </button>
  );
};
