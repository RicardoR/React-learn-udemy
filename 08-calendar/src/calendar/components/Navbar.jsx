export const Navbar = () => {
  return (
    <div className="navbar navbar-dark bg-dark mb-4 px-4">
      <div className="navbar-brand">
        <i className="fas fa-calendar-alt"></i>
        <span> Ricardo</span>
      </div>
      <button className="btn btn-outline-danger">
        <i className="fas fa-sign-out-alt"></i>
        <span>Salir</span>
      </button>
    </div>
  );
};
