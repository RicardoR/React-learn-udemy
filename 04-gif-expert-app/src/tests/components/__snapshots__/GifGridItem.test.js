import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../../components/GifGridItem';

describe('<GifGridItem />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<GifGridItem />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a card element', () => {
    expect(wrapper.find('.card')).toBeTruthy();
  });
});
