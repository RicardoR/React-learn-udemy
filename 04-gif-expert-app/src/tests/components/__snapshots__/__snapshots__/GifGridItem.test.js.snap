// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> should have a card element 1`] = `
<div
  className="card"
>
  <img />
  <p />
</div>
`;
